pkgs/firehose/lib/firehose.dart

```diff
@@ -130,7 +130,7 @@ Documentation at https://github.com/dart-lang/ecosystem/wiki/Publishing-automati
       print('changelog:');
       print(package.changelog.describeLatestChanges.trimRight());
 
-      if (pubspecVersion != changelogVersion) {
+      if (!changelogVersion.toString().contains(pubspecVersion)) {
```

Comment:
This will allow correct changes (`1.2.3` and `[1.2.3](...)`) but will also allow incorrect ones (`foo 1.2.3 bar`).

We should be more precise w/ this check. I think that either means updating `Changelog.latestVersion` to handle the two supported types of version formats, or changing `Changelog.latestVersion` to be named `Changelog.lastHeader` and having code which can understand that that is either a version or a linked version.

We should also have a test of the code that can parse `[1.2.3](...)` => `1.2.3`.

Reply:
No problem, I can add some test, but regarding the way to extract the number. Why would `foo 1.2.3 bar` wrong? Some repos meight use `Version: 1.2.3 [date]` or something similar as header. 

The only way this `contains` could go wrong is this dates. When the version is `26.5.2023`.

I'd recommend adding a regex match `[0-9]+.[0-9]+.[0-9]+` as `Changelog.latestVersion`. 
We could allow to change from the yaml it but I don't think that is neccesary.
What do you think?

Reply:
> Why would foo 1.2.3 bar wrong? Some repos meight use Version: 1.2.3 [date] or something similar as header.

I think I'd prefer to be proscriptive here - there's no hard format for a changelog; I'd just like to support a few clear, simple formats.

If you think you can pull the version out w/ a regex reliably (and have a few positive and negative tests to show that), sgtm.

Reply:
I've switched to using regex and added a lot of test cases, please let me know if that is what you had in mind